src/action-sheets/index.js

```diff
@@ -728,6 +728,9 @@ export const constructMessageActionButtons = (args: {|
   }
   if (message.sender_id === ownUser.user_id && messageNotDeleted(message)) {
     // TODO(#2793): Don't show if message isn't deletable.
+    // buttons.push(deleteMessage);
+  }
+  if (ownUser.role === 200 && messageNotDeleted(message)) {
     buttons.push(deleteMessage);
   }
```

Comment:
In general we try to avoid committing commented-out code, especially without a code comment explaining why and what it's about.

Do I understand correctly that you're trying to fix both #5528 and #4701 at the same time? I believe the way forward for the latter is https://github.com/zulip/zulip-mobile/issues/4701#issuecomment-1286196628:

> […]
> 
> * If you're an admin, you have permission.
> * Otherwise, if you didn't send the message, you don't have permission.
> * Otherwise (so, you sent the message but you're not an admin), it gets complicated -- you might have permission, and might not.
> 
> I think we can round off that third point to assuming you have permission, and showing the option in the UI. […]

The code in this revision doesn't match that:
- It should show the button for a non-admin user's own messages.
- The `messageNotDeleted` thing (checking if the message's content is `<p>(deleted)</p>`) should be irrelevant; probably that should be removed as part of dealing with #5528.

Also, if your commit is meant to fix two issues at once, please put a `Fixes: #…` line for both issues in the commit message, and also in the PR description: https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#github-prs-issues

Reply:
Ah, and there is a role that's even more powerful than an admin: an owner. When admins have permission to do something, owners should too.

For an example of another admin role check in this file, see `constructTopicActionButtons`:

```js
  if (roleIsAtLeast(ownUserRole, Role.Admin)) {
    buttons.push(deleteTopic);
  }
```